docs/jitserver_tuning.md

```diff
@@ -46,10 +46,10 @@ The client-session caches are deleted when the clients terminate, but this can h
 
  The maximum amount of memory that all the AOT cache instances combined can use at the server is 300 MB, by default. You can change this value by using the [`-XX:JITServerAOTmx=<size>`](xxjitserveraotmx.md) option. When the cache size reaches the specified limit, new clients cannot create new AOT cache instances or add new compiled methods to the existing AOT cache instances.
 
- Current limitations:
+ JITServer AOT cache can be reused by different instances of its JITServer server by using the [`-XX:+JITServerAOTCachePersistence`](xxjitserveraotcachepersistence.md) option. With this feature enabled, the JITServer server periodically saves its AOT caches to files. The server saves a new file when the number of AOT methods added to the in-memory cache since the last save equals or exceeds the number set in the `-Xjit:aotCachePersistenceMinDeltaMethods=<number of methods>` option. The default number is 200 methods. Even if the set number of methods is not reached, the server saves a new file if the time between two consecutive JITServer AOT cache save operations exceeds the time set in the `-Xjit:aotCachePersistenceMinPeriodMs=<milliseconds>` option. The default minimum time is 10000 milliseconds.
```

Comment:
So, if the number of methods is specified with the Xjit:aotCachePersistenceMinDeltaMethods=<number of methods>` option, then that number should be greater than 200 methods, right?

Also, if this number is specified, then the JITServer will need to add at least that specified number of methods to the in-memory cache for a save operation to occur, isn't it?

Is there a maximum number of methods that can be specified in this option?

Reply:
> if the number of methods is specified with the Xjit:aotCachePersistenceMinDeltaMethods=<number of methods> option, then that number should be greater than 200 methods, right?

Not necessarily. The JITServer saves its caches periodically, overwriting any existing cache files. Xjit:aotCachePersistenceMinDeltaMethods option is there to prevent saving too few extra methods, such that the overhead of saving could be greater than the advantage we can get from the extra methods.

> Also, if this number is specified, then the JITServer will need to add at least that specified number of methods to the in-memory cache for a save operation to occur, isn't it?

Yes. If the option is not specified, then this number is 200 extra methods.

> Is there a maximum number of methods that can be specified in this option?

No, as there is no need for such an option. See my explanation above why we want to wait for at least N methods before saving.